grails-app/services/org/broadinstitute/orsp/QueryService.groovy

```diff
@@ -725,14 +727,73 @@ class QueryService implements Status {
 
         def rows = getSqlConnection().rows(query, params)
         def ids = rows.collect{it.get("id")}
-        Collection result = Collections.emptyList()
+        Set result = new HashSet<Issue>()
 
         if (ids.size() > 0) {
-            result = Issue.findAllByIdInList(ids)
+            result = findIssuesSearchItemsDTO(ids)
         }
         result
     }
 
+    /**
+     * Find issues and create DTOs to narrow the amount of data used in the search process
+     * @param issueIds
+     * @return
+     */
+    Set<IssueSearchItemDTO> findIssuesSearchItemsDTO(ArrayList<Integer> issueIds) {
+        final String query = "SELECT i.id id, " +
+                "i.project_key projectKey, " +
+                "i.type type, " +
+                "i.status status, " +
+                "i.summary summary, " +
+                "i.reporter reporter, " +
+                "i.update_date updated, " +
+                "i.expiration_date expirationDate, " +
+                "iep.* " +
+                "FROM issue i LEFT JOIN issue_extra_property iep " +
+                "ON (iep.project_key = i.project_key AND iep.name in ('pm','pi','collaborator')) " +
+                "WHERE i.id IN (" + issueIds.join(",") + ")"
+
+        Set<IssueSearchItemDTO> resultDTO = new HashSet<IssueSearchItemDTO>()
+        IssueSearchItemDTO issueSearchItemDTO
+        String currentProjectKey = ""
+
+        getSqlConnection().rows(query).each {
+            if (it.get("projectKey") == currentProjectKey) {
+                if (it.get("type") != IssueType.CONSENT_GROUP.name) {
+                    def extraProp = issueSearchItemDTO.extraProperties.get(it.get("name").toString()) ?
+                            issueSearchItemDTO.extraProperties.get(it.get("name").toString()) : []
+                    extraProp.push(it.get("value").toString())
+                }
+            } else {
+                if (currentProjectKey != "") {
+                    resultDTO.add(issueSearchItemDTO)
+                }
+                currentProjectKey = it.get("projectKey")
+                issueSearchItemDTO = new IssueSearchItemDTO()
+
+                issueSearchItemDTO.setId((Integer)it.get("id"))
```

Comment:
I'm concerned about the casting here. We should strive mightily to avoid this. If you put this logic into the constructor, you could use a try block around a `Integer.valueOf()` line.